core/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -513,60 +502,59 @@ private void registerBuiltinFunctionScoreParsers() {
     }
 
     private void registerBuiltinQueryParsers() {
-        registerQuery(MatchQueryBuilder.PROTOTYPE::readFrom, MatchQueryBuilder::fromXContent, MatchQueryBuilder.NAME,
-                "match_phrase", "matchPhrase", "match_phrase_prefix", "matchPhrasePrefix", "matchFuzzy", "match_fuzzy", "fuzzy_match");
-        registerQueryParser(MultiMatchQueryParser::new);
-        registerQueryParser(NestedQueryParser::new);
-        registerQueryParser(HasChildQueryParser::new);
-        registerQueryParser(HasParentQueryParser::new);
-        registerQueryParser(DisMaxQueryParser::new);
-        registerQueryParser(IdsQueryParser::new);
-        registerQueryParser(MatchAllQueryParser::new);
-        registerQueryParser(QueryStringQueryParser::new);
-        registerQueryParser(BoostingQueryParser::new);
+        registerQuery(MatchQueryBuilder.PROTOTYPE::readFrom, MatchQueryBuilder::fromXContent, MatchQueryBuilder.NAME, "match_phrase",
+                "match_phrase_prefix", "match_fuzzy", "fuzzy_match");
+        registerQuery(MultiMatchQueryBuilder.PROTOTYPE::readFrom, MultiMatchQueryBuilder::fromXContent, MultiMatchQueryBuilder.NAME);
```

Comment:
Simplest thing is to make it a public static final string[] on the builder.
On Apr 1, 2016 8:22 AM, "Christoph Büscher" <notifications@github.com>
wrote:

> In core/src/main/java/org/elasticsearch/search/SearchModule.java
> <https://github.com/elastic/elasticsearch/pull/17458#discussion_r58197836>
> :
>
> > @@ -513,60 +502,59 @@ private void registerBuiltinFunctionScoreParsers() {
> >      }
> >
> >      private void registerBuiltinQueryParsers() {
> > -        registerQuery(MatchQueryBuilder.PROTOTYPE::readFrom, MatchQueryBuilder::fromXContent, MatchQueryBuilder.NAME,
> > -                "match_phrase", "matchPhrase", "match_phrase_prefix", "matchPhrasePrefix", "matchFuzzy", "match_fuzzy", "fuzzy_match");
> > -        registerQueryParser(MultiMatchQueryParser::new);
> > -        registerQueryParser(NestedQueryParser::new);
> > -        registerQueryParser(HasChildQueryParser::new);
> > -        registerQueryParser(HasParentQueryParser::new);
> > -        registerQueryParser(DisMaxQueryParser::new);
> > -        registerQueryParser(IdsQueryParser::new);
> > -        registerQueryParser(MatchAllQueryParser::new);
> > -        registerQueryParser(QueryStringQueryParser::new);
> > -        registerQueryParser(BoostingQueryParser::new);
> > +        registerQuery(MatchQueryBuilder.PROTOTYPE::readFrom, MatchQueryBuilder::fromXContent, MatchQueryBuilder.NAME, "match_phrase",
> > +                "match_phrase_prefix", "match_fuzzy", "fuzzy_match");
> > +        registerQuery(MultiMatchQueryBuilder.PROTOTYPE::readFrom, MultiMatchQueryBuilder::fromXContent, MultiMatchQueryBuilder.NAME);
>
> Can we move the whole names() method to the builder then (maybe slightly
> rename) and call it here when registering? That way we can later also
> include them in the QueryBuilder interface to enforce its implementation,
> and we can add tests for them.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/elastic/elasticsearch/pull/17458/files/092a90dc36ed1712d3d9c97e1f3eb6deba0e348e#r58197836>
>